client/client.go

```diff
@@ -233,3 +245,41 @@ func (c *Client) loadShortID() error {
 	c.shortID = binary.LittleEndian.Uint32(data)
 	return nil
 }
+
+// readCTFile looks for a CT settings file, and if it is found,
+// parses it by line for the energy multiplier and divider.
+func (c *Client) readCTSettingsFile() error {
+	path := filepath.Join(c.staticBaseDir, CTSettingsFile)
+	file, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil // use defaults
+		} else {
+			return fmt.Errorf("error opening ct settings file: %v", err)
+		}
+	}
+	defer file.Close()
+	var mult, div float64
+	scanner := bufio.NewScanner(file)
+	if scanner.Scan() {
+		line := scanner.Text()
+		mult, err = strconv.ParseFloat(line, 64)
+		if err != nil {
+			return fmt.Errorf("could not parse 1st ct settings line: %v", err)
+		}
+	} else {
+		return fmt.Errorf("ct settings file has no 1st line")
+	}
+	if scanner.Scan() {
+		line := scanner.Text()
+		div, err = strconv.ParseFloat(line, 64)
+		if err != nil {
+			return fmt.Errorf("could not parse 2nd ct settings line: %v", err)
+		}
+	} else {
+		return fmt.Errorf("ct settings file has no 2nd line")
+	}
```

Comment:
Similar here,  you can reduce nesting like so:

```
var line1, line2 string
if scanner.Scan() {
    line1 = scanner.Text()
 } else {
     // err
 }
 if scanner.Scan() {
     line2 = scanner.Text()
  } else {
      // err
  }
  
 mult, err = strconv.ParseFloat(line1, 64)
 if err != nil {
     // err
  }
  // same for div
  ```

Reply:
I wrote this too late at night lol